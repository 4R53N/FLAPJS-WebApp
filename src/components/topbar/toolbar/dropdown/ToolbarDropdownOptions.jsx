import React from 'react';
import PropTypes from 'prop-types';
import Style from './ToolbarDropdownOptions.module.css';

/**
 * A React component that can do anything you want. :D.
 */
class ToolbarDropdownOptions extends React.Component
{
    constructor(props)
    {
        super(props);
        this.state = {
            listOpen: false,
        };
        this.handleMouseOver = this.handleMouseOver.bind(this);
        this.handleMouseLeave = this.handleMouseLeave.bind(this);
    }

    handleMouseOver()
    {
        this.setState({
            listOpen: true
        });
    }

    handleMouseLeave()
    {
        this.setState({
            listOpen: false
        });
    }

    /** @override */
    render()
    {
        const props = this.props;
        const { listOpen } = this.state;

        return (
            <div
                id={props.id}
                style={props.style}
                // ref={node => this.node = node}
                className={Style.container}
                onBlur={() => this.handleMouseLeave()}
                onFocus={() => this.handleMouseOver()}
                onMouseOver={() => this.handleMouseOver()}
                onMouseLeave={() => this.handleMouseLeave()}
            >
                {/* <button className="dd-header" onClick={() => this.toggleList()}> */}
                <button>
                    <div>{props.title}</div>
                    {/* {listOpen ? <p>up</p> : <p>down</p>} */}
                </button>
                {listOpen &&
                    <ul className={Style.dropdownOptions}>
                        {props.list.map((item) => (
                            <li key={item.id}>
                                {
                                    item.options ?
                                        <ToolbarDropdownOptions
                                            title={item.title}
                                            list={item.options}
                                            toggleItem={props.toggleItem}
                                        />
                                        :
                                        <button
                                            onMouseDown={() => { props.toggleItem(item.id, item.key); }}>
                                            {item.title}
                                            {item.selected && <p>x</p>}
                                        </button>
                                }
                            </li>
                        ))}
                    </ul>}
            </div>
        );
    }
}

ToolbarDropdownOptions.propTypes = {
    id: PropTypes.string,
    style: PropTypes.object,
    className: PropTypes.string,
    title: PropTypes.string,
    list: PropTypes.array,
    toggleItem: PropTypes.func
};
ToolbarDropdownOptions.defaultProps = {
};

export default ToolbarDropdownOptions;
