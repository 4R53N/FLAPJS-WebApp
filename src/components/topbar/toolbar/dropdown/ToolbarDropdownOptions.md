# ToolbarDropdownOptions

## Motivation
Cause I love all of you. <3

## Work Log
- Created ToolbarDropdownOptions component.