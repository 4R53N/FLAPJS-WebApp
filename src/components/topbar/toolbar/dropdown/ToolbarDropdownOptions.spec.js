import React from 'react';
import { shallow } from 'enzyme';

import ToolbarDropdownOptions from './ToolbarDropdownOptions.jsx';

let wrapper;

beforeEach(() =>
{
    wrapper = shallow(<ToolbarDropdownOptions/>);
});

describe('<ToolbarDropdownOptions/> rendering', () =>
{
    test('renders correctly', () =>
    {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<ToolbarDropdownOptions/> interaction', () =>
{
    /** I promise I will test you soon... */
});

describe('<ToolbarDropdownOptions/> lifecycle', () =>
{
    /** ...later. */
});
