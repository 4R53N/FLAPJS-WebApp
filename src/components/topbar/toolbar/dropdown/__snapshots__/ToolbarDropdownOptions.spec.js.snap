// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolbarDropdownOptions/> rendering renders correctly 1`] = `
<div
  className="container"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <button>
    <div />
  </button>
</div>
`;
